### PR TITLE
Created Book entity, repository, services

### PR DESCRIPTION
**Pull request by vsych321**

### CODE REVIEW

**Review by evgenkolner — APPROVED** [review]
good decision

File: src/main/java/bookstore/entity/Book.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Table(name = "books")
+@Data
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/entity/Book.java:14]: Lombok is a great library but be [attentive when using it with Entity](https://jpa-buddy.com/blog/lombok-and-jpa-what-may-go-wrong/) JFYI

File: src/main/java/bookstore/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/repository/impl/BookRepositoryImpl.java]: As of Spring Framework 4.3, an @Autowired annotation on such a constructor is no longer necessary if the target bean defines only one constructor to begin with. See [note here ](https://docs.spring.io/spring-framework/reference/core/beans/annotation-config/autowired.html)Let’s check it ;)

File: src/main/java/bookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery(
+                    "select b from Book b", Book.class).getResultList();
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
                    "from Book", Book.class).getResultList();
```

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s fix a few moments;)

File: src/main/java/bookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery(
+                    "from Book b", Book.class).getResultList();
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/repository/impl/BookRepositoryImpl.java:44]:
```suggestion
                    "from Book", Book.class).getResultList();
```

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**